sympy/simplify/simplify.py

```diff
@@ -2540,8 +2541,8 @@ def recurse(arg, **kwargs):
 
     expr = sympify(expr)
 
-    if not isinstance(expr, Basic) or expr.is_Atom or expr in (
-            exp_polar(0), exp_polar(1)):
+    if (not isinstance(expr, Basic) or isinstance(expr, MatrixSymbol) or (
+            expr.is_Atom or expr in (exp_polar(0), exp_polar(1)))):
```

Comment:
With the changes from the last commit, is this still necessary? This is also a hack around strings being in MatrixSymbol, and won't properly traverse the tree in the args of `MatrixSymbols`.